src/cargo/core/features.rs

```diff
@@ -598,7 +598,7 @@ impl Features {
                 see_docs()
             ),
             Status::Unstable => {
-                if let Some(allow) = &config.cli_unstable().allow_features {
+                if let Some(allow) = allow_features {
```

Comment:
imo this change loses some important context of which set of allowed features this is.  This is specifically the `-Z` / CLI features.  Maybe `cli_allow_features` for the trait method and variable??